### PR TITLE
Form consistency changes per Form Guidelines design doc.

### PR DESCRIPTION
**Pull request by sg00dwin**
- Remove `.rbac-edit-binding` rules that are no longer needed.
- Move sub section headers above form fields (Role binding)
- Apply new class `. co-m-pane__form` to maintain consistent widths.
- Update styling of required field asterisk
- Added `.co-form-section__separator` to replace `<hr`> and generic `separator` class
- Apply `.btn-dropdown__item--placeholder` class to dropdown placeholder
- Add `co-required` icon where missing
- also set "Edit YAML" back to `font-weight:normal` since it was inheriting `font-weight: 300` from `<h1>`

Fixes https://jira.coreos.com/browse/CONSOLE-996

screen shot 2018-12-17 at 4 46 04 pm

---
screen shot 2018-12-17 at 4 28 25 pm

---
screen shot 2018-12-17 at 4 28 43 pm

---
screen shot 2018-12-17 at 4 43 54 pm

---
screen shot 2018-12-17 at 4 57 17 pm

### CODE REVIEW

**Comment by spadgett** [comment]
This looks a lot better!

I notice in a least one place, the dropdown is showing the default value as if it were placeholder text:

screen shot 2018-12-17 at 5 57 15 pm

**Comment by spadgett** [comment]
Buttons are a little close to the form inputs on the role binding form:

create role binding openshift 2018-12-17 18-01-28

**Comment by sg00dwin** [comment]
@spadgett updated pr for review

> Buttons are a little close to the form inputs on the role binding form:
> 
Added missing separator

screen shot 2018-12-18 at 12 16 54 pm

File: frontend/public/components/utils/dropdown.jsx [hunk 1]
```diff
@@ -446,6 +450,7 @@ export class ContainerDropdown extends React.PureComponent {
       items={dropdownItems}
       spacerBefore={spacerBefore}
       title={title}
+      noSelection={true}
```
[the remark this answers is not in the document]
Comment (sg00dwin) [reply frontend/public/components/utils/dropdown.jsx]: removed

File: frontend/public/components/utils/dropdown.jsx [hunk 2]
```diff
@@ -144,7 +144,11 @@ export class Dropdown extends DropdownMixin {
 
     this.state.items = Object.assign({}, bookmarks, props.items);
 
-    this.state.title = props.noSelection ? props.title : _.get(props.items, props.selectedKey, props.title);
+    // if noSelection is set to true then wrap title with span.btn-dropdown__item--placeholder
```
[the remark this answers is not in the document]
Comment (sg00dwin) [reply frontend/public/components/utils/dropdown.jsx]: removed

**Comment by spadgett** [comment]
jenkins rebuild

**Pull request merged**